homeassistant/components/isy994/helpers.py

```diff
@@ -320,44 +323,53 @@ def _categorize_nodes(
 
         if hasattr(node, "parent_node") and node.parent_node is None:
             # This is a physical device / parent node
-            hass_isy_data[ISY_DEVICES][node.address] = _generate_device_info(node)
-            hass_isy_data[ISY_ROOT_NODES][Platform.BUTTON].append(node)
+            isy_data.devices[node.address] = _generate_device_info(node)
+            isy_data.root_nodes[Platform.BUTTON].append(node)
+            # Any parent node can have communication errors:
+            isy_data.aux_props[Platform.SENSOR].append((node, PROP_COMMS_ERROR))
+            # Add Ramp Rate and On Levels for Dimmable Load devices
+            if getattr(node, "is_dimmable", False):
+                aux_controls = ROOT_AUX_CONTROLS.intersection(node.aux_properties)
+                for control in aux_controls:
+                    isy_data.aux_props[Platform.SENSOR].append((node, control))
 
         if node.protocol == PROTO_GROUP:
-            hass_isy_data[ISY_NODES][ISY_GROUP_PLATFORM].append(node)
+            isy_data.nodes[ISY_GROUP_PLATFORM].append(node)
             continue
 
         if node.protocol == PROTO_INSTEON:
             for control in node.aux_properties:
-                hass_isy_data[ISY_NODES][SENSOR_AUX].append((node, control))
+                if control in SKIP_AUX_PROPS:
+                    continue
+                isy_data.aux_props[Platform.SENSOR].append((node, control))
```

Comment:
Highlighting `#3` from the description -- Lines 326-344. As part of the structure change, `hass_isy_data[ISY_NODES][SENSOR_AUX]` has been moved out the root `isy_data` class since upcoming PRs will use more than the `SENSOR` platform.

As part of this move, also moved the filtering lines from `sensor.py` back to here.